code/ensemble/methods/create-ensemble-relative-skill.R

```diff
@@ -37,9 +37,14 @@ create_ensemble_relative_skill <- function(forecasts,
                                            average = "mean",
                                            by_horizon = FALSE,
                                            return_criteria = FALSE,
-                                           verbose = FALSE) {
+                                           verbose = FALSE,
+                                           ...) {
```

Comment:
I might be missing something but I don't think this one is needed.

Reply:
I've tried both ways but without the `...` I get an `unused argument` error which annoyingly stops the whole lot from running.

Reply:
To be honest this points to a problem to me as there I'm not clear which argument could be passed here that's known to `create_ensemble_relative_skill`. I'm probably missing something obvious but do you remember which argument the error was about?

Reply:
The unused argument was `exclude_designated_other`, which gets passed to `create_ensemble_relative_skill()` whenever it's run inside the wrapper of `run_ensemble(method = "relative_skill_")`.

`run_ensemble()` should include a way to pass on the necessary arguments for relative skill (`evaluation_date` and `continuous_weeks`), but I also think we shouldn't be forced to provide the args, because there are sensible defaults in `create_ensemble_relative_skill()` if they're missing. 

This can be achieved by passing along `...`, but that does then mean that  `create_ensemble_relative_skill()` has to accept all of whatever other general arguments are needed for the rest of `run_ensemble()`, as well as the specific args for this function. Hence if `create_ensemble_relative_skill()` doesn't have the `...`, I was getting the `unused argument` error.

I guess the alternative to really avoid the `...` in `create_ensemble_relative_skill()`, would be to grab the defaults for any missing args inside `run_ensemble()` before before it's run - like:

```
run_ensemble <- function(...) {
 extra_args <- list(...)
 # If necessary arguments not passed in through ... args, get function's defaults
    if (!"evaluation_date" %in% names(extra_args)) {
      evaluation_date = formals(create_ensemble_relative_skill)$evaluation_date
      }
 ensemble <- create_ensemble_relative_skill(evaluation_date)
```
Although this seems a bit more prone to failing if arguments change.

I might also be massively over-thinking this. I was trying to keep the ensemble code flexible so put it in this kind of nesting structure, but that may well have just over complicated it.

Reply:
Had a go at a simpler solution in https://github.com/epiforecasts/covid19-forecast-hub-europe/pull/609